assignments/prototypes.js

```diff
@@ -15,96 +15,143 @@
   * dimensions (These represent the character's size in the video game)
   * destroy() // prototype method that returns: `${this.name} was removed from the game.`
 */
+function GameObject(attrs){
+  this.createdAt = attrs.createdAt;
+  this.name = attrs.name;
+  this.dimensions = attrs.dimensions;
+}
+
+GameObject.prototype.destroy = function(){
+ return (`${this.name} was removed from the game.`);
+}
+
 
 /*
-  === CharacterStats ===
-  * healthPoints
-  * takeDamage() // prototype method -> returns the string '<object name> took damage.'
-  * should inherit destroy() from GameObject's prototype
+ === CharacterStats ===
+ * healthPoints
+ * takeDamage() // prototype method -> returns the string '<object name> took damage.'
+ * should inherit destroy() from GameObject's prototype
 */
 
+function CharacterStats(statAttrs){
+ GameObject.call(this, statAttrs);
+ this.healthPoints = statAttrs.healthPoints; 
+}
+CharacterStats.prototype = Object.create(GameObject.prototype);
+
+CharacterStats.prototype.takeDamage = function() {
+ return (`${this.name} took damage`); 
+}
+
+
+
+
 /*
-  === Humanoid (Having an appearance or character resembling that of a human.) ===
-  * team
-  * weapons
-  * language
-  * greet() // prototype method -> returns the string '<object name> offers a greeting in <object language>.'
-  * should inherit destroy() from GameObject through CharacterStats
-  * should inherit takeDamage() from CharacterStats
+ === Humanoid (Having an appearance or character resembling that of a human.) ===
+ * team
+ * weapons
+ * language
+ * greet() // prototype method -> returns the string '<object name> offers a greeting in <object language>.'
+ * should inherit destroy() from GameObject through CharacterStats
+ * should inherit takeDamage() from CharacterStats
 */
- 
+
+
+function Humanoid(humAttrs){
+ CharacterStats.call(this, humAttrs);
+ this.team = humAttrs.team;
+ this.weapons = humAttrs.weapons;
+ this.language = humAttrs.language;
+}
+
+Humanoid.prototype = Object.create(GameObject.prototype);
```

Comment:
Line 67 is unnecessary. Humanoid is a grand-child of GameObject and gains access to it's `__proto__` with line 68